### ISSUE TITLE
Declare arrays always empty to avoid failing when checking array size/count

### ISSUE THREAD

**Issue by NicoHood** [koalaman/shellcheck#2133]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
set -u

# Failing:
declare -a ARRAY

# Working:
# declare -a ARRAY=()

# Print array content
echo "${ARRAY[*]}"

# Check array count
if [[ "${#ARRAY[@]}" -eq 0 ]]; then
    echo "Empty array"
fi
```

#### Here's what shellcheck currently says:
* No issues

#### Here's what I wanted or expected to see:
When using `set -u`, arrays should be (in my opinion) declared empty. Otherwise the size check fails. For some reason printing the array content works, but checking the size dont. I am not sure if this is a bash bug though. However I might be wrong too.
